RevenueCatUI/Templates/Components/Stack/StackComponentView.swift

```diff
@@ -32,79 +38,131 @@ struct StackComponentView: View {
-                            )
-                        }
-                    case .flex:
-                        FlexVStack(
-                            alignment: horizontalAlignment.stackAlignment,
-                            spacing: viewModel.spacing,
-                            justifyContent: distribution.justifyContent,
-                            componentViewModels: self.viewModel.viewModels,
-                            onDismiss: self.onDismiss
-                        )
-                    }
-                }
+                VerticalStack(
+                    style: style,
+                    horizontalAlignment: horizontalAlignment,
+                    distribution: distribution,
+                    viewModels: self.viewModel.viewModels,
+                    onDismiss: self.onDismiss
+                )
             case .horizontal(let verticalAlignment, let distribution):
-                switch self.viewModel.hstackStrategy {
-                case .normal, .lazy:
-                    HStack(alignment: verticalAlignment.stackAlignment, spacing: viewModel.spacing) {
-                        ComponentsView(componentViewModels: self.viewModel.viewModels, onDismiss: self.onDismiss)
-                    }
-                case .flex:
-                    FlexHStack(
-                        alignment: verticalAlignment.stackAlignment,
-                        spacing: viewModel.spacing,
-                        justifyContent: distribution.justifyContent,
-                        componentViewModels: self.viewModel.viewModels,
-                        onDismiss: self.onDismiss
-                    )
-                }
+                HorizontalStack(
+                    style: style,
+                    verticalAlignment: verticalAlignment,
+                    distribution: distribution,
+                    viewModels: self.viewModel.viewModels,
+                    onDismiss: self.onDismiss
+                )
             case .zlayer(let alignment):
                 ZStack(alignment: alignment.stackAlignment) {
                     ComponentsView(componentViewModels: self.viewModel.viewModels, onDismiss: self.onDismiss)
                 }
             }
         }
-        .padding(viewModel.padding)
+        .padding(style.padding)
         .padding(additionalPadding)
-        .size(viewModel.size)
-        .background(viewModel.backgroundColor)
-        .shape(border: viewModel.border,
-               shape: viewModel.shape)
-        .applyIfLet(viewModel.shadow) { view, shadow in
+        .size(style.size)
+        .background(style.backgroundColor)
+        .shape(border: style.border,
+               shape: style.shape)
+        .applyIfLet(style.shadow) { view, shadow in
             // Without compositingGroup(), the shadow is applied to the stack's children as well.
             view.compositingGroup().shadow(shadow: shadow)
         }
-        .padding(viewModel.margin)
+        .padding(style.margin)
```

Comment:
how does this interact with line 75? Does it override it?

Reply:
@vegaro  This does look very unclear without context 😅  SwiftUI applies things on top of each other so...

1. Line 75 adds padding to the view to make it bigger
2. Line 78 adds background color (which is applied to the whole bigger view)
3. This line (85) adds `padding` again (but without background color) so it acts as margin